libs/net/core/Http/HttpRequestClient.cs

```diff
@@ -166,12 +166,12 @@ public virtual async Task<HttpResponseMessage> SendAsync(string url, HttpMethod?
                 }
             }
 
-            if (message.Headers.UserAgent == null)
+            if (string.IsNullOrEmpty(message.Headers.UserAgent.ToString()))
             {
                 message.Headers.Add("User-Agent", "TNO.Api");
             }
 
-            _logger.LogDebug("HTTP request made: {method}:{uri}", message.Method, message.RequestUri);
+            _logger.LogDebug($"HTTP request made: {message.Method}:{message.RequestUri} (User-Agent: {message.Headers.UserAgent?.ToString()})");
```

Comment:
This will give a lint error I believe.  Log messages are supposed to be parametrized.
`_logger.LogDebug("HTTP request made: {method}:{uri} (User-Agent: {agent})", message.Method, message.RequestUri, message.Headers.UserAgent?.ToString());`

Reply:
Updated, thanks